src/index.ts

```diff
@@ -1,58 +1,2 @@
-import { DependencyList, EffectCallback } from 'react';
-import * as React from 'react';
-
-const { useState, useEffect, useLayoutEffect } = React;
-
-type MediaQueryObject = { [key: string]: string | number | boolean };
-
-const camelToHyphen = (str: string) =>
-  str.replace(/[A-Z]/g, m => `-${m.toLowerCase()}`).toLowerCase();
-
-const objectToString = (query: string | MediaQueryObject) => {
-  if (typeof query === 'string') return query;
-  return Object.entries(query)
-    .map(([feature, value]) => {
-      feature = camelToHyphen(feature);
-      if (typeof value === 'boolean') {
-        return value ? feature : `not ${feature}`;
-      }
-      if (typeof value === 'number' && /[height|width]$/.test(feature)) {
-        value = `${value}px`;
-      }
-      return `(${feature}: ${value})`;
-    })
-    .join(' and ');
-};
-
-type Effect = (effect: EffectCallback, deps?: DependencyList) => void;
-const createUseMedia = (effect: Effect) => (
-  rawQuery: string | MediaQueryObject,
-  defaultState: boolean = false
-) => {
-  const [state, setState] = useState(defaultState);
-  const query = objectToString(rawQuery);
-  effect(() => {
-    let mounted = true;
-    const mql = window.matchMedia(query);
-    const onChange = () => {
-      if (!mounted) return;
-      setState(!!mql.matches);
-    };
-
-    mql.addListener(onChange);
-    setState(mql.matches);
-
-    return () => {
-      mounted = false;
-      mql.removeListener(onChange);
-    };
-  }, [query]);
-
-  return state;
-};
-
-
-export const useMedia = createUseMedia(useEffect);
-export const useMediaLayout = createUseMedia(useLayoutEffect);
-
-export default useMedia;
+export {default, useMediaLayout, mockMediaQueryList} from './useMedia';
+export {Effect, MediaQueryObject} from './types';
```

Comment:
What is the use case for exporting `mockMediaQueryList`, `Effect` and `MediaQueryObject`? Maybe we can export just the hooks from the top level here? And if somebody wants to import `mockMediaQueryList`, for example, they can still import from `useMedia.ts` file directly.

Reply:
I have reverted back to not exporting the `mockMediaQueryList` + `types` at this entry point.

I have also updated the README to show a use case _(in my app at least)_ for using the `mockMediaQueryList`.

Let me know if you are okay with the latest commits.